### ISSUE TITLE
order-in-components has issues

### ISSUE THREAD

**Issue by RafLeszczynski** [netguru/eslint-plugin-netguru-ember#3]
I've experiences issues following issues with `order-in-components` rule:

- fails for properties like:
```JS
Component.extend({
  // ArrayExpression
  classNames: ['aaa', 'bbb'],
  classNameBindings: ['ccc', 'ddd'],

  // ObjectExpression
  someObj: {},
});
```
`isDefaultProp` only checks for `literals`

- component life cycle hooks are not checked as they are `FunctionExpression` and return `null` when it comes to order value

- same goes for any other custom / private functions

- services are treated the same as single line computed - `CallExpression` with line length of `1 `

- same goes to multiline computed and observers - no distinction 

I'm happy to fix all these issues with PR but first I'd like to set the order as current order is equivalent to netguru guidelines so any changes here will probably require to change the guidelines if you want to make it consistent.

My order proposal:
1. default props (Literal, Arrays or Objects)
2. services
3. single line computed
4. multiline computed
5. observers
6. hooks
7. actions
8. custom / private methods.

 @michalsnik what do you think about it?

**Comment by michalsnik** [comment]
Hi @RafLeszczynski, thanks again for reaching out,

I experienced similar problem recently - just haven't got time to take a deeper look in this.

I like your proposal, and I'm willing to update our styleguide to reflect this order.

But I'd also would like to know what do you think about this @jniechcial ? Do you agree with me that this looks reasonable and we should go with this one?

**Comment by jniechcial** [comment]
Looks very reasonable, thanks for your input @RafLeszczynski!

**Comment by RafLeszczynski** [comment]
Cool I'll try to find some time during next weekend and prepare PR. 

BTW I've introduced this plugin into our ESLint config @ [AirHelp](https://www.airhelp.com). There are couple other issues that I found so expect more raised issues and PR soon.

**Comment by michalsnik** [comment]
Great, glad to hear that! We're open for any suggestions and PR's :)

**Comment by RafLeszczynski** [comment]
Please take a look at this PR: https://github.com/netguru/eslint-plugin-netguru-ember/pull/4

**Comment by RafLeszczynski** [comment]
As agreed in separate discussion the changed order should look like this:

1. services
2. default props (Literal, Arrays or Objects)
3. single line computed
4. multiline computed
5. observers
6. hooks
7. actions
8. custom / private methods.

**Comment by michalsnik** [comment]
@RafLeszczynski Thanks for your commitment and updating our plugin, we really appreciate it! :)

I already released it under `0.0.6` 🎉 

Btw. feel free to check my PR with some adjustments according to another scenario that might occur.
